File: articles/on_java_exceptions/propagate.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	if parsed, err := myFunc(); err != nil {
		fmt.Println(err)
	} else {
		fmt.Println(parsed)
	}
}

func myFunc() (int, error) {
	if parsed, err := strconv.Atoi("someVal"); err != nil {
		return 0, err
	} else {
		return parsed, nil
	}
}

File: articles/on_java_exceptions/propagate.rs
use std::fs::File;
use std::io::Result;

fn main() {
    let _ = match my_func() {
        Ok(file) => file,
        Err(error) => {
            panic!("Problem opening the file: {:?}", error)
        },
    };
}

fn my_func() -> Result<File> {
    let file = File::open("hello.txt")?;
    print!("Successfully opened file {:?}", file);
    Ok(file)
}
